From 71c21f61469edc1efc98e3fb58f3413141d2c6db Mon Sep 17 00:00:00 2001
From: Henry Chen <chenx97@aosc.io>
Date: Tue, 23 Sep 2025 10:07:50 +0800
Subject: [PATCH] AOSCOS: HID: core: cap mouse polling rate to 500Hz.

The RPI kernel patch claims that capping the polling rate to 60Hz reduces
overhead when using X, but this breaks mouse usability on platforms that
use non-xhci drivers.

RPI devices using xhci are simply too powerful to care about reducing polling
rates, while RPI3 uses dwc_otg and polling mouse inputs at 60Hz is
unacceptable.
---
 drivers/hid/usbhid/hid-core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/hid/usbhid/hid-core.c b/drivers/hid/usbhid/hid-core.c
index fd1a33087978..75df81e1d788 100644
--- a/drivers/hid/usbhid/hid-core.c
+++ b/drivers/hid/usbhid/hid-core.c
@@ -1117,8 +1117,8 @@ static int usbhid_start(struct hid_device *hid)
 		 */
 		switch (hid->collection->usage) {
 		case HID_GD_MOUSE:
-			if (hid_mousepoll_interval == ~0 && interval < 16)
-				interval = 16;
+			if (hid_mousepoll_interval == ~0 && interval < 2)
+				interval = 2;
 			else if (hid_mousepoll_interval != ~0 && hid_mousepoll_interval != 0)
 				interval = hid_mousepoll_interval;
 			break;
-- 
2.51.0

